File: 1493. Longest Subarray of 1's After Deleting One Element/1493. Longest Subarray of 1's After Deleting One Element.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int n = nums.size();
        int maxlen=0, len=0, len1=0, len2=0;
        for (int i=0; i<n; ++i) {
            if (nums[i] > 0) {
                ++len1;
            } else {
                maxlen = maxlen < len1 + len2 ? len1 + len2 : maxlen;
                len2 = len1;
                len1 = 0;
            }
        }
        maxlen = maxlen < len1 + len2 ? len1 + len2 : maxlen;
        maxlen = maxlen == n ? n - 1 : maxlen;
        return maxlen;
    }
};

File: 1493. Longest Subarray of 1's After Deleting One Element.py
# time - O(n), place - O(1)

class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        l = 0
        last_zero = -1
        max_length = 0
        for i in range(len(nums)):
            if nums[i] == 0:
                l = last_zero + 1
                last_zero = i
            max_length = max(max_length, i - l)

        return max_length